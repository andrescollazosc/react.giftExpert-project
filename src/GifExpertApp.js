import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GitfGrid } from "./components/GitfGrid";

const GifExpertApp = ({ defaultCategories =[] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  console.log(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category, i) => {
          return <GitfGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};

export default GifExpertApp;
