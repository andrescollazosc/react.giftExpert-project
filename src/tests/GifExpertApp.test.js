import React from "react";
import GifExpertApp from "../GifExpertApp";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Testing in <GifExpertApp />", () => {
  test("should be show successfuly ", () => {
    const wrapper = shallow(<GifExpertApp/>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Must be show one list of categories", () => {
    const categories = ["One punch", "Dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GitfGrid").length).toBe(categories.length);
  });
});
