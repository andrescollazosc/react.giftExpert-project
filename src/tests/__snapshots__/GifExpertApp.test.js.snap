// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifExpertApp /> Must be show one list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GitfGrid
      category="One punch"
      key="One punch"
    />
    <GitfGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Testing in <GifExpertApp /> should be show successfuly  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
